x-pack/solutions/observability/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -405,22 +421,45 @@ function getErrorCountByParentId(errorDocs: TraceAPIResponse['traceItems']['erro
   }, {});
 }
 
-export function getOrphanItemsIds(waterfall: IWaterfallSpanOrTransaction[]) {
+export function getOrphanItemsIds(
+  waterfall: IWaterfallSpanOrTransaction[],
+  entryWaterfallTransactionId?: string
+) {
   const waterfallItemsIds = new Set(waterfall.map((item) => item.id));
   return waterfall
-    .filter((item) => item.parentId && !waterfallItemsIds.has(item.parentId))
+    .filter(
+      (item) =>
+        // the first item should never be orphan
+        entryWaterfallTransactionId !== item.id &&
+        item.parentId &&
+        !waterfallItemsIds.has(item.parentId)
+    )
     .map((item) => item.id);
 }
 
 export function reparentOrphanItems(
   orphanItemsIds: string[],
   waterfallItems: IWaterfallSpanOrTransaction[],
-  newParentId?: string
+  entryWaterfallTransaction?: IWaterfallTransaction
 ) {
   const orphanIdsMap = new Set(orphanItemsIds);
+
   return waterfallItems.map((item) => {
     if (orphanIdsMap.has(item.id)) {
-      item.parentId = newParentId;
+      if (
+        entryWaterfallTransaction &&
+        (item.duration > entryWaterfallTransaction.duration ||
+          item.doc.timestamp.us < entryWaterfallTransaction.doc.timestamp.us)
+      ) {
+        // we need to hide the orphan item if it's longer or if it has started before the entry transaction
+        // as this means it's a parent of the entry transaction
```

Comment:
Q: what's the implication of this on the waterfall?

Reply:
Can't we just filter this item out from the list?